### PR TITLE
Refactor RestClientConfiguration and DefaultRestClientTest

### PR DESCRIPTION
**Pull request by banterCZ**
- Remove no-arg constructor from `RestClientConfiguration`.
- Remove public from `DefaultRestClientTest`.

### CODE REVIEW

File: rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java [hunk 1]
```diff
@@ -30,12 +29,6 @@
  */
 public class RestClientConfiguration {
 
-    /**
-     * Constructor of REST client configuration.
-     */
-    public RestClientConfiguration() {
```
Comment (banterCZ) [inline rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java:36]: Useless no-arg constructor, same as the default one.

File: rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public class DefaultRestClientTest {
     private static final int SYNCHRONIZATION_TIMEOUT = 10000;
 
     @BeforeEach
-    public void initRestClient() throws RestClientException {
```
Comment (banterCZ) [inline rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java:68]: Not needed for JUnit 5, less is more.

File: rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java (hunk 1 above)
Comment (romanstrobl) [reply rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java:36]: Please do not remove the no-arg constructor. Jackson and similar frameworks require a no-arg constructor to function properly. We do not serialize the configuration into JSON right now but we might one day.

File: rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java (hunk 2 above)
Comment (romanstrobl) [reply rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java:68]: Good idea.

**Review by romanstrobl — CHANGES_REQUESTED** [review]
I am not convinced about removal of no-arg constructor, because this would break Jackson serialization, should we decide to serialize the configuration. Other than this looks good.

File: rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java (hunk 1 above)
Comment (romanstrobl) [reply rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java:36]: I take it back... there is no other constructor. Ok, we can remove it.

Comment (banterCZ) [reply rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java:36]: There is no other constructor, so default makes the same result.

**Review by romanstrobl — APPROVED** [review]
Approved.

File: rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java (hunk 1 above)
Comment (romanstrobl) [reply rest-client-base/src/main/java/com/wultra/core/rest/client/base/RestClientConfiguration.java:36]: I see, ok, let's remove it then.

**Pull request merged**